shared/app/nav.native.js

```diff
@@ -42,6 +49,16 @@ class CardStackShim extends Component {
     }
   }
 
+  shouldComponentUpdate(nextProps: CardStackShimProps) {
+    return (
+      this.props.mode !== nextProps.mode ||
```

Comment:
Another option here is to use shallowEquals which let's you have specific handlers for certain keys